test/controllers/LogoutControllerSpec.scala

```diff
@@ -44,21 +44,25 @@ class LogoutControllerSpec extends BaseSpec {
         session(result).get(SessionKeys.token) mustBe empty
 
       }
-      "when already logged out" in {
 
-        val controller: LogoutController = inject[LogoutController]
+      "when already logged out" in new LogoutTestCase {
         val result = controller
           .logout()
           .apply(FakeRequest())
 
         status(result) mustBe SEE_OTHER
         redirectLocation(result) mustBe Some(loginPagePath)
         session(result).get(SessionKeys.token) mustBe empty
-
       }
 
     }
 
   }
 
+  class LogoutTestCase() {
```

Comment:
This might be of general value in the `BaseControllerSpec`.

Reply:
I view it as a context for this suite only. I am against adding things to the BaseControllerSpec because:
* I don't want to pollute the namespace for all tests
* I think that mocks will conflict if we set up mocks in the base controller class. Mockito is hard enough to reason about is. I think each test should be responsible for setting up the class that it's testing. In this case, no mocks are being set up, but I imagine they will for the other controllers.

Reply:
OK.